sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -595,6 +595,19 @@ class HiveDDLSuite
     }
   }
 
+  test("test show partitions") {
+    val message = intercept[AnalysisException] {
+      sql("SHOW PARTITIONS default.nonexistentTable")
+    }.getMessage
+    assert(message.contains("Table does not exist"))
+
+    withTable("t1") {
+      sql("CREATE TABLE t1 (key STRING, value STRING) PARTITIONED BY (ds STRING)")
+      sql("ALTER TABLE t1 ADD PARTITION (ds = '1')")
+      assert(sql(" SHOW PARTITIONS t1").schema.getFieldIndex("partition") == Some(0))
```

Comment:
My original reasoning for adding this test was  to exercise the code change that makes the output attribute reference name for SHOW PARTITIONS command to be ‘partition’.  This test covers this scenario.    Without the fix, it was ‘result’.  With the fix it is ‘partition’.

But it turns out another PR already made the code change for it. But no tests were added in that pr.  So I have kept this as a separate test for now.  

If you prefer we do not need to have this test, we can remove it.